xbmc/ServiceManager.h

```diff
@@ -0,0 +1,43 @@
+/*
+ *      Copyright (C) 2005-2016 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include "addons/AddonManager.h"
+#include "addons/BinaryAddonCache.h"
+#include <memory>
+
+namespace ADDON {
+class CAddonMgr;
+class CBinaryAddonCache;
+}
+
+class CServiceMgr
```

Comment:
Could we rename this to `CServiceManager`? I've always found this very confusing about `CAddonMgr` because I have to include `addons/AddonManager.h` and then I expect the class to be named `CAddonManager` like with all other classes where the name of the class matches the name of the header file (prepended with a `C`) but then it doesn't compile because the class name is actually `CAddonMgr`.